### DISCUSSION TITLE
How to run a dns server on hostNetwork for external requests?

### DISCUSSION THREAD

**Discussion by mi-bauer** [siderolabs/talos#9921] [Q&A]
I need to run talos on a single node in an offline environment. Besides its main workload that node should manage a local network so that it is possible to connect to the node locally. I already got dhcp running on the hostNetwork (its on the hostNetwork to also enable access to the talos and kubernetes APIs from that network). Now I want to run a dns server that provides a few dns entries to external requests that target the node IP. However port 53 seems to be used already by the DNS caching server. How can I keep that DNS caching server running and still use port 53 for external requests? Any help is appreciated. :)

**Comment by smira** [comment]
That's not true:

```
tcp     0        0        127.0.0.53:53        0.0.0.0:*         LISTEN
udp     0        0        127.0.0.53:53        0.0.0.0:*
```

Talos listens only on specific addresses, so you can still bind to :53 with proper IP.

You can also disable host DNS if you feel it would help you.
